fract4d/c/fract4dc/calcargs.cpp

```diff
@@ -17,30 +18,36 @@ calc_args::calc_args()
     params = new double[N_PARAMS];
 }
 
+static std::mutex ref_count_mutex;
+
 void calc_args::set_cmap(PyObject *pycmap_)
 {
     pycmap = pycmap_;
     cmap = colormaps::cmap_fromcapsule(pycmap);
+    const std::lock_guard<std::mutex> lock(ref_count_mutex);
```

Comment:
I feel like the ideal pattern would be if only the main thread ever dealt with Python refcounts and all the calc threads could ignore it. But haven't thought much about how complicated that would be.

Reply:
I’ve been playing around with GIL acquisition to try to address the last threadsanitizer report. No luck so far.
I guess the GTK signals triggering the calculation process out of the interpreter thread had something to do.

Reply:
interrupted should most probably be std::atomic<> and that makes the mutex and data race go away. Volatile is not the right language feature in this case as it doesn't mean thread safe and only causes the compiler to always emit read instructions when the variable is referenced, it doesn't make the compiler ensure cache cohearency or atomicity.